### ISSUE TITLE
[Windows] App freezes when using destroy()

### ISSUE THREAD

**Issue by 7Eltantawy** [leanflutter/window_manager#502]
- Dependency version 
`window_manager: ^0.4.2`

- Flutter Version:
```
Flutter 3.24.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 2663184aa7 (5 weeks ago) • 2024-09-11 16:27:48 -0500
Engine • revision 36335019a8
Tools • Dart 3.5.3 • DevTools 2.37.3
```

When I try to close the app and accept closing from dialog
the app freeze for a while and then close

I have tried to remove the dialog and everything around but the problem seems to be from `destroy()` function

Here is my implementation
```dart
  @override
  void initState() {
    asyncInit();
    super.initState();
  }

  Future asyncInit() async {
    if (!PlatformExtension.isDesktop) return;
    await windowManager.setPreventClose(true);
    windowManager.addListener(this);
    await windowManager.setAlwaysOnTop(sl<SettingsRepo>().getPinWindows()); 
  }

  @override
  Future onWindowClose() async {
    final bool isPreventClose = await windowManager.isPreventClose();
    if (!isPreventClose) return;
    if (!mounted) return;

    final result = await showAskExitDialog(context);
    if (result == null || !result) return;

    await windowManager.destroy();
  }
  
  @override
  void dispose() {
    if (PlatformExtension.isDesktop) {
      windowManager.removeListener(this);
    }
    super.dispose();
  }
```

But when do the following it close immediately
```dart
  @override
  Future onWindowClose() async {
    final bool isPreventClose = await windowManager.isPreventClose();
    if (!isPreventClose) return;
    if (!mounted) return;

    final result = await showAskExitDialog(context);
    if (result == null || !result) return;

    await windowManager.setPreventClose(false);
    await windowManager.close();
  }
```

**Comment by AminBhst** [comment]
Same issue here on same version of Flutter. For me, even using `windowManager.close();` doesn't fix the issue. The app still freezes for a second or two before closing.
I don't think this is a bug that was introduced by window_manager itself. I have reverted the window_manager version to a previous one that I'm sure didn't suffer from this problem. My guess is that this is due to something that was introduced by the Flutter SDK which is not being properly accounted for in window_manager.

**Comment by AminBhst** [comment]
Same issue here. I have tried downgrading the Flutter SDK to versions `3.24.0`, `3.22.0`, `3.19.0` and the issue persisted. I rolled back my Flutter version to the one I remember didn't suffer from this problem which is `3.19.6` and it was fixed. So my guess is that there were some SDK changes that are not being properly accounted for in window_manager. For me, the problem isn't fixed by `windowManager.close();` either.
This is a big issue that I had to give up on some new Flutter features only to prevent this from happening.
